talk/morelanguage/exceptions.tex

```diff
@@ -4,185 +4,172 @@
-    \item even throw() (no exceptions)
-    \end{itemize}
+   \end{itemize}
   \end{block}
-  \pause
-  \begin{exampleblock}{long live noexcept}
-    \begin{itemize}
-    \item noexcept a somehow equivalent to throw()
-    \item but is checked at compile time
-    \item so allows compiler optimizations
-    \end{itemize}
-  \end{exampleblock}
+  \begin{multicols}{2}
+    \begin{cppcode*}{fontsize=\tiny,gobble=2}
+      // try ... catch in hot loop
+      //   might impact performance
+      for (const &item: list) {
+        try {
+          check_item(item); // Can throw bad_item
+          ... // process item
+        }
+        catch (bad_item &e) {
+          ... // ignore this item, but keep going
+        }
+      }
+    \end{cppcode*}
+    \columnbreak
+    \begin{cppcode*}{fontsize=\tiny,gobble=2}
+      // For this code logic C-style error codes
+      //   may be better
+      for (const &item: list) {
+        auto err_code = check_item(item);
+        if (!err_code) {
+          ... // process item
+        } else {
+          ... // warn and carry on
+        }
+      }
+    \end{cppcode*}
+  \end{multicols}
 \end{frame}
 
+
 \begin{frame}[fragile]
-  \frametitlecpp[11]{Full power of noexcept}
-  \begin{block}{3 uses of noexcept}
+  \frametitlecpp[11]{noexcept}
+  \begin{block}{noexcept}
     \begin{itemize}
-    \item standalone
-      \begin{cppcode*}{gobble=2, linenos=false}
+      \item specifying the \textit{noexcept} specifier states that a function is guaranteed throw an exception
+      \begin{cppcode*}{fontsize=\scriptsize,gobble=2,linenos=false}
         int f() noexcept;
       \end{cppcode*}
-    \item as an expression saying whether exceptions can be sent
-      \begin{cppcode*}{gobble=2, linenos=false}
-        int f() noexcept(sizeof(long) == 8);
-      \end{cppcode*}
-    \item as an operator to know whether a function launches exceptions
-      \begin{cppcode*}{gobble=2, linenos=false}
-        constexpr bool callCannotThrow = noexcept(f());
-        if constexpr (callCannotThrow) { ... }
-      \end{cppcode*}
-    \item and you can combine them
-      \begin{cppcode*}{gobble=2, linenos=false}
-        template <typename T> void foo()
-             noexcept(noexcept(T())) {}
-      \end{cppcode*}
-   \end{itemize}
+      \begin{itemize}
+        \item either no exceptions will be thrown or they are handled internally
+        \item checked at compile time, so it allows the compiler to optimise around that knowledge
+      \end{itemize}
+      \item \textit{noexcept(expression)} specifier evaluates the expression and, if \textit{true}, the function won't throw
+        \begin{cppcode*}{fontsize=\scriptsize,gobble=2,linenos=false}
+          int safe_if_long_is_8_bytes() noexcept(sizeof(long)==8);
+        \end{cppcode*}
+      \item Use \textit{noexcept} on leaf functions where you can be sure of the behaviour
+      \item Note that since C++11 destructors are \textit{noexcept} - never try and throw from them
```

Comment:
```suggestion
      \item Note that since C++11 destructors are \textit{noexcept} - never throw from them
```
I missinterpreted your sentence as: I should not use a `try` block in a dtor, which you can use.